examples/complete/main.tf

```diff
@@ -1,3 +1,101 @@
-# Complete example
+# Resource group
 ##############################################################################
+
+module "resource_group" {
+  source  = "terraform-ibm-modules/resource-group/ibm"
+  version = "1.1.0"
+  # if an existing resource group is not set (null) create a new one using prefix
+  resource_group_name          = var.resource_group == null ? "${var.prefix}-resource-group" : null
+  existing_resource_group_name = var.resource_group
+}
+
+##############################################################################
+# Key Protect All Inclusive
+##############################################################################
+
+locals {
+  key_ring_name = "en-key-ring"
+  key_name      = "${var.prefix}-en"
+}
+module "key_protect_all_inclusive" {
+  source                    = "terraform-ibm-modules/key-protect-all-inclusive/ibm"
+  version                   = "4.2.0"
+  resource_group_id         = module.resource_group.resource_group_id
+  region                    = var.region
+  key_protect_instance_name = "${var.prefix}-kp"
+  resource_tags             = var.resource_tags
+  key_map = {
+    (local.key_ring_name) = [local.key_name]
+  }
+}
+
+##############################################################################
+# Get Cloud Account ID
+##############################################################################
+
+data "ibm_iam_account_settings" "iam_account_settings" {
+}
+
+##############################################################################
+# VPC
+##############################################################################
+resource "ibm_is_vpc" "example_vpc" {
+  name           = "${var.prefix}-vpc"
+  resource_group = module.resource_group.resource_group_id
+  tags           = var.resource_tags
+}
+
+resource "ibm_is_subnet" "testacc_subnet" {
+  name                     = "${var.prefix}-subnet"
+  vpc                      = ibm_is_vpc.example_vpc.id
+  zone                     = "${var.region}-1"
+  total_ipv4_address_count = 256
+  resource_group           = module.resource_group.resource_group_id
+}
+
+##############################################################################
+# Create CBR Zone
+##############################################################################
+
+module "cbr_zone" {
+  source           = "terraform-ibm-modules/cbr/ibm//modules/cbr-zone-module"
+  version          = "1.15.1"
+  name             = "${var.prefix}-VPC-network-zone"
+  zone_description = "CBR Network zone representing VPC"
+  account_id       = data.ibm_iam_account_settings.iam_account_settings.account_id
+  addresses = [{
+    type  = "vpc",
+    value = ibm_is_vpc.example_vpc.crn
+  }]
+}
+
+module "event_notification" {
+  source                    = "../../"
+  resource_group_id         = module.resource_group.resource_group_id
+  name                      = "${var.prefix}-en"
+  kms_encryption_enabled    = true
+  existing_kms_instance_crn = module.key_protect_all_inclusive.key_protect_id
```

Comment:
Is the `key_protect_id` output the CRN value?

Reply:
Yes
<img width="1187" alt="image" src="https://github.com/terraform-ibm-modules/terraform-ibm-event-notifications/assets/4889052/6b1ec6d7-c37d-446a-95e2-94f9e0a19be2">